src/KnownRelaysManager.ts

```diff
@@ -0,0 +1,292 @@
+      if (!genericEvent) {
+        log.info('No event with suitable args found');
+
+        return;
+      }
+
+      const {
+        args: { relayManager },
+      } = genericEvent;
+
+      if (!genericEvent) {
+        log.info('No relayManager found in event');
+
+        return;
+      }
+
+      foundRelayManagers.add(relayManager);
+    });
+
+    log.info(
+      `fetchRelaysAdded: found unique relays: ${JSON.stringify(
+        Array.from(foundRelayManagers.values())
+      )}`
+    );
+    this.latestScannedBlock = toBlock;
+
+    return foundRelayManagers;
+  }
+
+  _refreshFailures(): void {
+    const newMap = new Map<string, RelayFailureInfo[]>();
+    this.relayFailures.forEach((value: RelayFailureInfo[], key: string) => {
+      newMap.set(
+        key,
+        value.filter((failure) => {
+          const elapsed = (new Date().getTime() - failure.lastErrorTime) / 1000;
+
+          return elapsed < this.config.relayTimeoutGrace;
+        })
+      );
+    });
+    this.relayFailures = newMap;
+  }
+
+  async getRelaysSortedForTransaction(
+    transactionDetails: EnvelopingTransactionDetails
+  ): Promise<IRelayHub.RelayManagerDataStruct[][]> {
+    return [
+      this.preferredRelayers as IRelayHub.RelayManagerDataStruct[],
+      await this._sortRelaysInternal(transactionDetails, this.allRelayers),
+    ];
+  }
+
+  async _sortRelaysInternal(
+    transactionDetails: EnvelopingTransactionDetails,
+    activeRelays: IRelayHub.RelayManagerDataStruct[]
+  ): Promise<IRelayHub.RelayManagerDataStruct[]> {
+    const scores = new Map<string, number>();
+    for (const activeRelay of activeRelays) {
+      let score = 0;
+      score = await this.scoreCalculator(
+        this.relayFailures.get(await activeRelay.url) ?? [],
+        activeRelay,
+        transactionDetails
+      );
+      scores.set(await activeRelay.manager, score);
+    }
+
+    const resolvedActiveRelays = await Promise.all(
+      activeRelays.map(async (activeRelay) => {
+        const { currentlyStaked, manager, registered, url } = activeRelay;
+
+        return {
+          currentlyStaked: await currentlyStaked,
+          manager: await manager,
+          registered: await registered,
+          url: await url,
+        };
+      })
+    );
+
+    return Array.from(resolvedActiveRelays.values()).sort((a, b) => {
+      const aScore = scores.get(a.manager) ?? 0;
+      const bScore = scores.get(b.manager) ?? 0;
+
+      return bScore - aScore;
+    });
+  }
+
+  saveRelayFailure(
+    lastErrorTime: number,
+    relayManager: string,
+    relayUrl: string
+  ): void {
+    const relayFailures = this.relayFailures.get(relayUrl);
+    const newFailureInfo = {
+      lastErrorTime,
+      relayManager,
+      relayUrl,
+    };
+    if (!relayFailures) {
+      this.relayFailures.set(relayUrl, [newFailureInfo]);
+    } else {
+      relayFailures.push(newFailureInfo);
+    }
+  }
+
+  getLastScannedBlock(): number {
```

Comment:
It looks like this method isn't used anywhere

Reply:
removed this along with the private member latestScannedBlock, which strangely isn’t used anywhere.